### PR TITLE
[WIP] [SPARK-4286] Add dynamic allocation to the Spark coarse-grained schedule...

### PR DESCRIPTION
**Pull request by dragos**
## Don't merge, testing the build bot.

Based on #3861.

### CODE REVIEW

**Comment by dragos** [comment]
retest this please

File: core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala [hunk 1]
```diff
@@ -156,7 +164,7 @@ private[spark] class CoarseMesosSchedulerBackend(
       command.setValue(
         "%s \"%s\" org.apache.spark.executor.CoarseGrainedExecutorBackend"
           .format(prefixEnv, runScript) +
-        s" --driver-url $driverUrl" +
+        s" --driver-url $driverURL" +
```
Comment (tnachen) [inline core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala]: Through out spark code base URLs are usually spelled Url.

**Comment by tnachen** [comment]
LGTM

File: core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala (hunk 1 above)
Comment (dragos) [inline core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala]: Ok, I'll change that, thanks.

**Pull request closed without merging**